Documentation: Make the SerenityOS Patterns documentation more idiomatic

ninadsachania opened the pull request.

nico commented on Documentation/Patterns.md:
```diff
@@ -164,6 +165,7 @@ Instead, `serenity_main(..)` is defined like this:
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
+    // code
```
Same.

nico commented on Documentation/Patterns.md:
```diff
@@ -3,22 +3,22 @@
 ### Examples:
 
-Example from LibGfx:
+Example from [LibGfx](../Userland/Libraries/LibGfx/Bitmap.cpp):
```
Let's not link to actual code. It adds close to nothing, and adds the risk of the link becoming stale.

ninadsachania replied:
Done.

nico commented on Documentation/Patterns.md:
````diff
@@ -39,7 +39,7 @@ ErrorOr<NonnullRefPtr<Bitmap>> Bitmap::create_shareable(BitmapFormat format, Int
 ```
 
-Example from the Kernel:
+Example from the [Kernel](../Kernel/Memory/AddressSpace.cpp):
````
same

nico commented on Documentation/Patterns.md:
```diff
@@ -130,30 +130,31 @@ private:
 int main(int argc, char** argv)
 {
-   return 0;
+    // code
```
This line adds nothing, remove it.

nico commented on Documentation/Patterns.md:
I think LHS is supposed to be title case. Leave these alone.

nico commented on Documentation/Patterns.md:
```diff
@@ -303,23 +304,23 @@ static StringView example_fn(const SourceLocation& loc = SourceLocation::current
 }
 
-int main(int, char**)
+int main(void)
```
Undo this change. (writing an explicit `void` in a parameter list is C, where functions without any parameters are implicitly varargs. In C++, it's `main()`, but the LHS is fine too. The RHS is wrong.)

nico reviewed (commented):
Borders on frivolous, but I suppose some of these are kinda better. Others are worse though, and the sum total benefit has a hard time to be worth the time spent on it. Againm, I'd recommend working on less superficial things.

(Still, I looked at it, and if the next revision addresses comments, I'll hit merge.)

ninadsachania commented:
> Borders on frivolous, but I suppose some of these are kinda better. Others are worse though, and the sum total benefit has a hard time to be worth the time spent on it. Again, I'd recommend working on more substantive things.
> 
> (Still, I looked at it, and if the next revision addresses comments, I'll hit merge.)

Thanks for reviewing!

I did create PRs with more substantive improvements, but nobody is reviewing them.

Here are some examples:

+ https://github.com/SerenityOS/serenity/pull/25435
+ https://github.com/SerenityOS/serenity/pull/25405
+ https://github.com/SerenityOS/serenity/pull/25388

nico commented:
> I did create PRs with more substantive improvements, but nobody is reviewing them.

They are also < 10 lines of code touched – _even more_ substantive! (I felt like I said this last time too, and then kind of disengaged when the message didn't seem to be heard.)

ninadsachania commented:
> > I did create PRs with more substantive improvements, but nobody is reviewing them.
> 
> They are also < 10 lines of code touched – _even more_ substantive! (I felt like I said this last time too, and then kind of disengaged when the message didn't seem to be heard.)

I am confused by what you mean by "substantive"? Should I create PRs that touch lots of lines?

Two of PRs that I linked above are bug fixes. I don't know how to make them more substantive than that.

ninadsachania commented on Documentation/Patterns.md:
```diff
@@ -319,7 +318,7 @@ Example Usage:
 
 #if LOCK_DEBUG
-#    include <AK/SourceLocation.h>
+    #include <AK/SourceLocation.h>
```
No, it's now new. That was a mistake on my part.
